Source/drlg_l1.cpp

```diff
@@ -2048,30 +2038,50 @@ void CryptFloor(int rndper)
 bool PlaceCathedralStairs(lvl_entry entry)
 {
 	bool success = true;
+	Point position;
+	Displacement offset { 3, 4 };
 
 	// Place poison water entrance
 	if (Quests[Q_PWATER].IsAvailable()) {
-		if (!PlaceMiniSet(PWATERIN, entry == ENTRY_RTNLVL))
+		Point miniPosition;
+		if (!PlaceMiniSet(PWATERIN, &miniPosition)) {
 			success = false;
-		if (entry == ENTRY_RTNLVL)
-			ViewPosition += Displacement { 2, 3 };
+		} else {
+			int8_t t = TransVal;
+			TransVal = 0;
+			DRLG_MRectTrans(miniPosition.x, miniPosition.y + 2, miniPosition.x + 5, miniPosition.y + 4);
+			TransVal = t;
+			Displacement entranceOffset { 5, 7 };
+			Quests[Q_PWATER].position = miniPosition * 2 + entranceOffset + Displacement { 16, 16 };

```

Comment:
This comes up a lot, how about a helper in point.cpp :D

```cpp
Point Point::miniToWorld() {
    return {16 + 2 * x, 16 + 2 * y};
}
```

(been meaning to do a PR for this for a while)

Reply:
Confusingly minisets are constructed out of mega tiles :)